Add coco-caption with inheritance from coco-detection

koshtiakanksha opened the pull request:
Issue #141
Related to PR #163

cregouby commented on R/dataset-coco-caption.R:
```diff
@@ -0,0 +1,101 @@
+#' }
+#'
+#' @details
+#' This dataset inherits from `coco_detection_dataset` and reuses most of its logic,
+#' except it loads image-caption pairs instead of bounding boxes and object labels.
```
**todo** please remove. This is developper stuf, end user don't care.

cregouby commented on R/dataset-coco-caption.R:
```diff
@@ -0,0 +1,101 @@
+#'
+#' Loads the Microsoft COCO dataset for image captioning.
+#'
+#' @param root Root directory for the dataset.
+#' @param train Logical. Whether to load the training split (`TRUE`) or validation split (`FALSE`). Default is `TRUE`.
+#' @param year Year of the dataset. Only `"2014"` is currently supported.
+#' @param download If `TRUE`, downloads the dataset if it's not already present.
```
**improvement** you can also inherit params ! see #165

cregouby reviewed (commented):
**praise**  This is a huge improvement, with lower effort and reduced download and storage footprint.
**todo** to bypass the import issue, please move this code into the existing `cooco_dataset.R` file. and remove the ImportFrom.

cregouby commented on R/dataset-coco-caption.R:
```diff
@@ -0,0 +1,101 @@
+#' title(main = caption, col.main = "black", font.main = 1)
+#' }
+#'
+#' @importFrom torchvision coco_detection_dataset
```
**todo** remove

koshtiakanksha commented:
@cregouby 
Added coco_caption_dataset alongside coco_detection_dataset for centralized logic. Removed internal dev notes and reused param documentation via inheritance. Let me know if anything needs further cleanup!
